### ISSUE TITLE
About writing a imagepy plugin. (take the liberty to invite you)

### ISSUE THREAD

**Issue by yxdragon** [DeepLabCut/DeepLabCut#171]
Hi, I saw DeepLabCut form image.sc 's title icon, so I find this repo in github. The demo gif movies are so cool, And I am excited when I saw deeplabcut is in pure python and has a wx ui interface. 

May I take the liberty to invite you. I wish we can collaborate to write a deeplabcut plugins for [ImagePy](https://github.com/Image-Py/imagepy), which is a image process framework like ImageJ, but implemented in pure python, and with a wx ui. 

In ImagePy we can use any method and libraries based on Numpy directly. And the interaction such as label, image io, image view, marker is ready-made. And we can show chart (loss function curve), and table(net and trainning result) easily in ImagePy. So I think there will be no trouble to integrate. I can do a start.

lastly, a question, when I try "pip install deeplabcut", it erase so many packages to old version, I think it is not friendly for a developer, and break some environment dependence. Is this needed? could it be solved?

I think both ImagePy and DeepLabCut would benefit from a plugin easy to install and friendly to use. Anyone interested in?

**Comment by AlexEMG** [comment]
Your pitch sounds interesting, let us know how you proceed!

**Comment by yxdragon** [comment]
I had done some plugins based on tensorflow and keras befor (just predict, not train). I want make it useful for no-programmer, So two quesstions:
1. could deeplabcut be installed easily? (It seems not difficult, but we need gpu to train online, I think we can, and we can install package with pip in ImagePy's ui)
2. How long does the training cost? If it need more than 30 minute, I think It is not very useful for no-programmer scientist.

If a no-programmer can install easily, and label the image(in ImagePy), then got the result soon, and test it, If there are unsatisfies, repair lable, train again. (may be a little like ilastik), I think it would be very usable. And we can start it. I can do all the interaction and ui layer with a demo as a start.

**Comment by AlexEMG** [comment]
1. The installation is relatively straightforward and non-programmers manage to do it, e.g. with conda environments: https://github.com/AlexEMG/DeepLabCut/tree/master/conda-environments
2. Training time is usually .5- 24h (depending on the frame size etc.) This "long" time is justified, because most users want train a tailored network for their experiments, see e.g. our original study for our reasoning https://arxiv.org/abs/1804.03142

**Comment by MMathisLab** [comment]
I will close this issue for now, but feel free to comment on it!
